src/Microsoft.AspNetCore.Diagnostics/DeveloperExceptionPage/Views/ErrorPage.cshtml

```diff
@@ -39,17 +40,30 @@
                     location = firstFrame.Function;
                 }
             }
-            if (!string.IsNullOrEmpty(location) && firstFrame != null && !string.IsNullOrEmpty(firstFrame.File))
-            {
+                if (!string.IsNullOrEmpty(location) && firstFrame != null && !string.IsNullOrEmpty(firstFrame.File))
+                {
                 <p class="location">@location in <code title="@firstFrame.File">@System.IO.Path.GetFileName(firstFrame.File)</code>, line @firstFrame.Line</p>
+                }
+                else if (!string.IsNullOrEmpty(location))
+                {
+                <p class="location">@location</p>
+                }
+                else
+                {
+                <p class="location">@Resources.ErrorPageHtml_UnknownLocation</p>
+                }
             }
-            else if (!string.IsNullOrEmpty(location))
+        @{ var outerException = Model.ErrorDetails.Last(); }
+        @if(outerException?.Error is ReflectionTypeLoadException)
+        {
+            var exception = outerException.Error as ReflectionTypeLoadException;
+            if (exception.LoaderExceptions.Count() > 0)
```

Comment:
It is possible to create an instance with zero length. But not sure if this will ever be zero in a common scenario.